### ISSUE TITLE
un-compilable due soup newer api

### ISSUE THREAD

**Issue by mckaygerhard** [lpereira/hardinfo#48]
since the commit 907a6249077745aa6d19eb7280823dd769223a29 i cannot compile.. 

the errors its due https://developer.gnome.org/libsoup/stable/libsoup-2.4-XMLRPC-Support.html#soup-xmlrpc-parse-response are only available since soup 2.52 

i understand due since 2.52 the `soup_xmlrpc_extract_method_response` are marked DEPRECATED in flavor of `soup_xmlrpc_parse_response` so the commit update the hardinfo compilation process to up to date soup api.. 

**maybe sound complicated but this made hardinfo unnavailable to compile for some**  that still uses the old soup api.. 

as i remenber some specific versions of hardware does not install with newer versions of software, as example VIA unichrome GPU does not offerts OpenGL capabilities with recent Xorg versions.. due not DRI v1 capabilities.. and yes i have a VIA motherboard and only works with some specific xorg version and specific linux kernel, some other examples are Asus Eeepc or Geode Daruma (yes also have one of that)

**Comment by mckaygerhard** [comment]
the error:

```
libhardinfo-shell.a(syncmanager.c.o): In function `_soup_get_xmlrpc_value_int':
syncmanager.c:(.text+0x1a5): undefined reference to `soup_xmlrpc_parse_response'
libhardinfo-shell.a(syncmanager.c.o): In function `_soup_get_xmlrpc_value_string':
syncmanager.c:(.text+0x2c9): undefined reference to `soup_xmlrpc_parse_response'
libhardinfo-shell.a(syncmanager.c.o): In function `_soup_xmlrpc_call':
syncmanager.c:(.text+0x376): undefined reference to `soup_xmlrpc_message_new'
libhardinfo-shell.a(syncmanager.c.o): In function `_soup_xmlrpc_call_with_parameters':
syncmanager.c:(.text+0x473): undefined reference to `soup_xmlrpc_build_request'
libhardinfo-shell.a(syncmanager.c.o): In function `sync_dialog_start_sync':
syncmanager.c:(.text+0x970): undefined reference to `soup_session_new_with_options'
```

**Comment by mckaygerhard** [comment]
umm after a investigation there's no api way compatible solution, must be a if-else conditional.. depending of the version detected must be used some calls.. but i not find how detect the version way of soup! the macro `SOUP_CHECK_VERSION` are only available since 2.42 and must compile agains 2.24, someone have some tips?
a trick can be detecting older glib, due newe soup seem only compiles with newer glib, as i can see in https://git.gnome.org/browse/libsoup/tree/configure.ac#n81 minimum now are 2.38 glib so i can detect glib if are too newer i can use the soup check version, assuming are newer too, if not, then "else" to use the older soup api...

**Comment by bp0** [comment]
This is the case on Raspbian. To build, just don't have libsoup-dev installed (or disable it).
The feature is not working anyway, as xmlrpc.hardinfo.org does not respond to the requests.

**Comment by vegnuli** [comment]
I re-test compilation witout and post feedback here

**Comment by bp0** [comment]
PR #64 adds a cmake option to disable libsoup even if it is present.

**Comment by mckaygerhard** [comment]
due PR #64 closed this issue

**Comment by mckaygerhard** [comment]
reopens due PR #66

**Comment by mckaygerhard** [comment]
now closed due merged #66
